orchagent/copporch.cpp

```diff
@@ -0,0 +1,540 @@
+            }
+            else if (fvField(*i) == copp_policer_pir_field)          
+            {
+                attr.id = SAI_POLICER_ATTR_PIR;
+                attr.value.u64 = std::stoul(fvValue(*i));
+                policer_attribs.push_back(attr);
+                SWSS_LOG_DEBUG("obtained pir:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_action_green_field)
+            {
+                sai_packet_action_t policer_action;
+                if (!getPacketAction(fvValue(*i), policer_action))
+                {
+                    SWSS_LOG_ERROR("Invalid green action specified %s\n", fvValue(*i).c_str());
+                    return task_process_status::task_invalid_entry;
+                }
+                attr.id = SAI_POLICER_ATTR_GREEN_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_red_field) 
+            {
+                sai_packet_action_t policer_action;
+                if (!getPacketAction(fvValue(*i), policer_action))
+                {
+                    SWSS_LOG_ERROR("Invalid red action specified %s\n", fvValue(*i).c_str());
+                    return task_process_status::task_invalid_entry;
+                }
+                attr.id = SAI_POLICER_ATTR_RED_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_yellow_field)
+            {
+                sai_packet_action_t policer_action;
+                if (!getPacketAction(fvValue(*i), policer_action))
+                {
+                    SWSS_LOG_ERROR("Invalid yellow action specified %s\n", fvValue(*i).c_str());
+                    return task_process_status::task_invalid_entry;
+                }
+                attr.id = SAI_POLICER_ATTR_YELLOW_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown copp field specified:%s\n", fvField(*i).c_str());
+                return task_process_status::task_invalid_entry;
+            }
+        }
+
+        if (m_trap_map.find(key) != m_trap_map.end())
+        {
+            SWSS_LOG_DEBUG("found existing trap_group object:%s", key.c_str());
+            if(!policer_attribs.empty())
+            {
+                attr.id = SAI_HOSTIF_TRAP_GROUP_ATTR_POLICER;
+                attr.value.oid = SAI_NULL_OBJECT_ID;
+                if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->get_trap_group_attribute(m_trap_map[key], 1, &attr)))
+                {
+                    SWSS_LOG_ERROR("Failed to get policer for trap gorup:%s, error:%d\n", key.c_str(), sai_status);
+                    return task_process_status::task_failed;
+                }
```

Comment:
why do you need to retrieve the policer id from the SAI. Whether or not you have created one, your app should know.